src/util/misc.ts

```diff
@@ -161,3 +162,113 @@ export function dedup<T>(arr: T[]): T[] {
 
     return res;
 }
+
+/**
+ * Build a flattened version of the array of arrays `arr` and return it.
+ */
+export function flatten<T>(arr: Iterable<T[]>): T[] {
+    const res: T[] = [];
+    for (const el of arr) {
+        res.push(...el);
+    }
+
+    return res;
+}
+
+/**
+ * Given a list of T's `things` and a partial ordering between them `order` return
+ * a topologically sorted version of `things`. For any pair `[a,b]` in `order` we assume
+ * that `a` has to come before `b`.
+ */
+export function topoSort<T>(things: T[], order: Array<[T, T]>): T[] {
+    const successors = new Map<T, Set<T>>();
+    const nPreds = new Map<T, number>();
+
+    // Initialize datastructures
+    for (const thing of things) {
+        nPreds.set(thing, 0);
+        successors.set(thing, new Set());
+    }
+
+    for (const [a, b] of order) {
+        nPreds.set(b, (nPreds.get(b) as number) + 1);
+        (successors.get(a) as Set<T>).add(b);
+    }
+
+    const res: T[] = [];
+    for (const thing of things) {
+        if ((nPreds.get(thing) as number) === 0) {
+            res.push(thing);
+        }
+    }
+
+    assert(res.length > 0, `Order ${pp(order)} is not a proper partial order`);
+    let i = 0;
+
+    while (res.length < things.length) {
+        const curLength = res.length;
+        for (; i < curLength; i++) {
+            for (const successor of successors.get(res[i]) as Set<T>) {
+                const newCount = (nPreds.get(successor) as number) - 1;
+                nPreds.set(successor, newCount);
+
+                if (newCount === 0) {
+                    res.push(successor);
+                }
+            }
+        }
+
+        assert(
+            res.length > curLength,
+            `Order ${pp(order)} is not a valid proper order. Topo sort stalled at ${
+                res.length
+            } out of ${things.length}`
+        );
+    }
+
+    return res;
+}
+
+export function zip<T1, T2>(a1: readonly T1[], a2: readonly T2[]): Array<[T1, T2]> {
+    assert(
+        a1.length === a2.length,
+        `Mismatch in length between ${pp(a1)} of len ${a1.length} and ${pp(a2)} of len ${a2.length}`
+    );
+
+    const res: Array<[T1, T2]> = [];
+    for (let i = 0; i < a1.length; i++) {
+        res.push([a1[i], a2[i]]);
+    }
+
+    return res;
+}
+
+const writersCache = new Map<string, ASTWriter>();
+
+export function print(n: ASTNode, version = "0.8.0"): string {
+    let writer = writersCache.get(version);
+    if (writer === undefined) {
+        writer = new ASTWriter(DefaultASTWriterMapping, new PrettyFormatter(4), "0.8.0");
```

Comment:
We better pass `version` as a third argument instead of raw `"0.8.0"`. Also consider using [`LatestCompilerVersion`](https://github.com/ConsenSys/solc-typed-ast/blob/master/src/compile/constants.ts#L75) instead of `"0.8.0"` as default value for the argument.

Or... Is it a workaround to upgrade code to `0.8.0`?

Reply:
Changed it to `LatestCompilerVersion` - no this is just for debugging error output. 
Fixed in 35d34c0.